File: src/main/java/org/stt/event/EventBusModule.java
package org.stt.event;

import dagger.Module;
import dagger.Provides;
import net.engio.mbassy.bus.MBassador;

import javax.inject.Singleton;
import java.util.logging.Logger;

@Module
public abstract class EventBusModule {
    private static final Logger LOG = Logger.getLogger(EventBusModule.class.getSimpleName());

    private EventBusModule() {
    }

    @Provides
    @Singleton
    static MBassador<Object> provideMBassador() {
        return new MBassador<>(error -> LOG.severe(error::toString));
    }
}

File: src/main/kotlin/org/stt/event/EventBusModule.kt
package org.stt.event

import dagger.Module
import dagger.Provides
import net.engio.mbassy.bus.MBassador
import java.util.logging.Logger
import javax.inject.Singleton

@Module
class EventBusModule {
    private val log = Logger.getLogger(EventBusModule::class.java.simpleName)

    @Provides
    @Singleton
    fun provideMBassador(): MBassador<Any> {
        return MBassador { error -> log.severe { error.toString() } }
    }
}
